### ISSUE TITLE
localhost not being assigned to correct group

### ISSUE THREAD

**Issue by muratdx** [ansible/ansible#16882]
##### ISSUE TYPE

- Bug Report
##### ANSIBLE VERSION

```
ansible-playbook 2.1.1.0
  config file =
  configured module search path = Default w/o overrides
```
##### OS / ENVIRONMENT

CentOS Linux release 7.2.1511 (Core)
##### SUMMARY

In 2.1.1.0, "localhost" no longer appears to be assigned to the group it is listed under in the inventory. Instead, it is being placed in "ungrouped." This is resulting in some variables being inaccessible. This was working properly in 2.1.0.0 and prior releases.
##### STEPS TO REPRODUCE

```
Inventory:

[group1]
localhost ansible_connection=local

[group1:vars]
var1=100


Playbook:

- hosts: localhost
  tasks:
        - debug: var=group_names
        - debug: var=hostvars['localhost']['group_names']
        - debug: var=var1
        - debug: var=hostvars['localhost']['var1']


```

##### EXPECTED RESULTS

This is the output seen in 2.1.0.0:

```
PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [debug] *******************************************************************
ok: [localhost] => {
    "group_names": [
        "group1"
    ]
}

TASK [debug] *******************************************************************
ok: [localhost] => {
    "hostvars['localhost']['group_names']": [
        "group1"
    ]
}

TASK [debug] *******************************************************************
ok: [localhost] => {
    "var1": 100
}

TASK [debug] *******************************************************************
ok: [localhost] => {
    "hostvars['localhost']['var1']": "100"
}
```
##### ACTUAL RESULTS

Here is output from 2.1.1.0:

```
PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [debug] *******************************************************************
ok: [localhost] => {
    "group_names": [
        "group1"
    ]
}

TASK [debug] *******************************************************************
ok: [localhost] => {
    "hostvars['localhost']['group_names']": [
        "ungrouped"
    ]
}

TASK [debug] *******************************************************************
ok: [localhost] => {
    "var1": 100
}

TASK [debug] *******************************************************************
ok: [localhost] => {
    "hostvars['localhost']['var1']": "VARIABLE IS NOT DEFINED!"
}

```

**Comment by winggundamth** [comment]
I confirmed this. This happen to me too after upgrade to version 2.1.1.0.
I suspect this commit cause this error https://github.com/ansible/ansible/commit/5f1bbb4fcdea21b106367a5a426dc6f2b4851b9c. I still couldn't find any document about this change but is this intents?

**Comment by winggundamth** [comment]
ping @jimi-c

**Comment by bcoca** [comment]
can you check against devel? my tests indicate this was fixed by issue merged above

**Comment by jimi-c** [comment]
I just tested this against devel, and it appears f7570f1dc4c75795f48c0fe0fff1edb467f8df2f did fix this, so we'll go ahead and close it now. This will also be included in the next release (2.1.2 and/or 2.2.0).

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!
